TidepoolOnboarding/View Models/OnboardingViewModel.swift

```diff
@@ -215,29 +215,43 @@ class OnboardingViewModel: ObservableObject, CGMManagerOnboarding, PumpManagerOn
 
     func verifyDevice(completion: @escaping (OnboardingError?) -> Void) {
         guard deviceValid == nil else {
-            log.debug("%{public}@ device was already validated. deviceValid: %{public}@", #function, deviceValid! ? "Yes" : "No")
+            log.info("%{public}@ device was already validated. deviceValid: %{public}@", #function, deviceValid! ? "Yes" : "No")
             completion(nil)
             return
         }
 
         guard let tidepoolService = tidepoolService else {
-            log.debug("%{public}@ tidepool service does not exist", #function)
+            log.info("%{public}@ tidepool service does not exist", #function)
             completion(OnboardingError.unexpectedState)
             return
         }
 
-        // If the device does not require verification (i.e. simulator) or DeviceCheck is not supported, just mark as valid
-        guard deviceRequiresVerification, DCDevice.current.isSupported else {
-            log.debug("%{public}@ device requires verification: %{public}@; is device supported: %{public}@", #function, deviceRequiresVerification ? "Yes" : "No", DCDevice.current.isSupported ? "Yes" : "No")
+        // If the device does not require verification (i.e. simulator), mark as valid
+        guard deviceRequiresVerification else {
+            log.info("%{public}@ device does not require verification: %{public}@; is device supported: %{public}@", #function, deviceRequiresVerification ? "Yes" : "No", DCDevice.current.isSupported ? "Yes" : "No")
             self.deviceValid = true
             completion(nil)
             return
         }
 
+        // if the device does not support DCDevice API, mark as invalid
+        guard DCDevice.current.isSupported else {
+            log.info("%{public}@ DCDevice API is not supported. Without this API, the device token cannot be generated and this device automatically fails validation", #function)
+            self.deviceValid = false
+            completion(nil)
+            return
+        }
+
+        guard !ValidDeviceDetector.isJailbrokenDevice() else {
```

Comment:
Consider moving this above the `guard DCDevice.current.isSupported else` since we'd still want to have this check even if the DCDevice API isn't supported.